### PR TITLE
Replace Vagrant with Docker (Compose)

### LINKED ISSUE #39: Update Vagrant box to ubuntu/bionic64
Trusty (14.04) is EoL. Move to latest LTS, Bionic (18.04).

A problem is that we only support MySQL 5.5/5.6, while Bionic ships with MySQL 5.7.

See https://github.com/twingly/ecco/blob/e83cbfcffbb09833f42d6bf570ccd0c0ae903407/Vagrantfile#L4

### PR DESCRIPTION
**Pull request by walro**
The Vagrant box we used is outdated. Also, I'd argue that docker is a tad easier to maintain.

Close #39

### CODE REVIEW

**Comment by walro** [comment]
(Test failures obviously unrelated)

Edit: Addressed in e0808b6

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+version: "3.1"
+
+services:
+  db:
+    image: mysql:5.6
+    volumes:
+      - ./docker/ecco-my.cnf:/etc/mysql/conf.d/ecco-my.cnf
+    ports:
+      - 3306:3306
+    command: --default-authentication-plugin=mysql_native_password
```
Comment (walro) [inline docker-compose.yml]: Hm, this should already be the default. I will try and drop it on Monday and try again. I used https://docs.docker.com/samples/library/mysql/#-via-docker-stack-deploy-or-docker-compose as a starting point.

Comment (walro) [reply docker-compose.yml]: https://dev.mysql.com/doc/refman/5.6/en/authentication-plugins.html states it's the default. For 5.7 and onwards that statement is gone, probably why it was explicit in the example.

Comment (walro) [reply docker-compose.yml]: 402b2ae

**Review by dentarg — APPROVED** [review]

**Comment by dentarg** [comment]
Ran `docker-compose up` and `bin/all_specs` without any problems 👍

**Comment by dentarg** [comment]
Use Docker at Travis?

**Comment by dentarg** [comment]
Ah, maybe out of scope for this PR :-)

**Comment by walro** [comment]
> Ah, maybe out of scope for this PR :-)

Yeah, think so. But good idea, I'll look into it.

**Pull request merged**